homeassistant/components/saj/sensor.py

```diff
@@ -0,0 +1,195 @@
+                if (sensor.per_day_basis and date.today() > sensor.date_updated) or (
+                    not sensor.per_day_basis and not sensor.per_total_basis
+                ):
+                    task = sensor.async_update_values(unkown_state=True)
+                    if task:
+                        tasks.append(task)
+            if tasks:
+                await asyncio.wait(tasks)
+
+            return
+
+        values = await saj.read(used_sensors)
+        if not values:
+            try:
+                backoff = [1, 1, 1, 6, 30][backoff_step]
+                backoff_step += 1
+            except IndexError:
+                backoff = 60
+            return
+        backoff_step = 0
+
+        for sensor in hass_sensors:
+            task = sensor.async_update_values()
+            if task:
+                tasks.append(task)
+        if tasks:
+            await asyncio.wait(tasks)
+
+    interval = config.get(CONF_SCAN_INTERVAL) or timedelta(seconds=5)
+    async_track_time_interval(hass, async_saj, interval)
+
+
+class SAJsensor(Entity):
+    """Representation of a SAJ sensor."""
+
+    def __init__(self, pysaj_sensor):
+        """Initialize the sensor."""
+        self._sensor = pysaj_sensor
+        self._state = self._sensor.value
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"saj_{self._sensor.name}"
+
+    @property
+    def icon(self):
+        """Return the icon for the sensor."""
+        return SENSOR_ICON
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._sensor.unit
+
+    @property
+    def should_poll(self) -> bool:
+        """SAJ sensors are updated & don't poll."""
+        return False
+
+    @property
+    def per_day_basis(self) -> bool:
+        """Return if the sensors value is on daily basis or not."""
+        return self._sensor.per_day_basis
+
+    @property
+    def per_total_basis(self) -> bool:
+        """Return if the sensors value is cummulative or not."""
+        return self._sensor.per_total_basis
+
+    @property
+    def date_updated(self) -> date:
+        """Return the date when the sensor was last updated."""
+        return self._sensor.date
+
+    def async_update_values(self, unkown_state=False):
+        """Update this sensor."""
+        update = False
+
+        if self._sensor.value != self._state:
+            update = True
+            self._state = self._sensor.value
+
+        if unkown_state and self._sensor.value != STATE_UNKNOWN:
+            update = True
+            self._state = STATE_UNKNOWN
+
+        return self.async_update_ha_state() if update else None
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this sensor."""
+        return f"saj-{self._sensor.key}-{self._sensor.name}"
```

Comment:
We don't need to include `"saj"` in the unique_id. The entity registry is aware of the integration domain and the platform domain.

How stable are sensor.key and sensor.name? Are they changeable by the user somehow?

Reply:
Thanks for the information!
sensor.key and sensor.name are actually very stable and not changeable by the user.
I'll change this to `self._sensor.name` since it's unique on its own.